src/tests/run.proj

```diff
@@ -412,6 +470,10 @@ namespace $([System.String]::Copy($(Category)).Replace(".","_").Replace("\","").
              Properties=""
              Condition=" '$(MonoAotCompile)'=='true' " />
 
+    <MSBuild Projects="$(MSBuildProjectFile)"
+             Targets="BuildAndroidApp"
+             Condition=" '$(RunWithAndroid)'=='true' " />
```

Comment:
I believe that the below clause <code>_CMDDIR=%(TestDirectories.Identity)</code> is what makes MSBuild trigger the build multiple times, once for each item in the <code>TestDirectories</code> item group. Could you try to put a similar clause on the BuildAndroidApp target invocation to see whether it helps? Alternatively the %() clause can be used directly in the BuildAndroidApp target definition similar to what happens here:

https://github.com/dotnet/runtime/blob/b75f8d9fd1b3bcac5d82469bf39f1aea2c3a1652/src/tests/Common/helixpublishwitharcade.proj#L132

Reply:
Thanks for the suggestion. The issue has been resolved.